### PR TITLE
Do not fail job on failed get_job_id_url

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
We won't fail the job when GH fails to retrieve the job ID and URLs.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 1641c4ff8d6a7fdd40d12745d9a39893756634b4 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62651/1641c4ff8d6a7fdd40d12745d9a39893756634b4/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/env_helper.py [hunk 1]
```diff
@@ -42,23 +42,35 @@
...
 
-def GITHUB_JOB_URL() -> str:
-    GITHUB_JOB_ID()
+def GITHUB_JOB_URL(safe: bool = True) -> str:
+    try:
+        GITHUB_JOB_ID()
+    except APIException:
+        if safe:
+            logging.warning("Using run URL as a fallback to not fail the job")
+            return GITHUB_RUN_URL
```
Comment (maxknv) [inline tests/ci/env_helper.py:68]: there should be a raise in "non-safe" mode

**Review by maxknv — COMMENTED** [review]
Aside from the change you did already, i have 2 other suggestions:

1. decrease the number of job items in GH response here: https://github.com/ClickHouse/ClickHouse/blob/9cbfcf47630bf3c0944d7725c147f51e2452f512/tests/ci/env_helper.py#L90C49-L90C61
it's set to 100 now, and i noticed that if GH fails with this number it works fine with smaller number
2. do not repeat getting job id GH API call in case of failure (for safe mode or even for any mode)

**Comment by Felixoid** [comment]
> 1\. decrease the number of job items in GH response

It will drastically increase the number of requests. Thinking of "how to keep GH API from overload" is not our business.



> 2\. do not repeat getting job id GH API call in case of failure

Why would we? No, let's retry to get the best result. And if GH API would be fixed in between - good. No - well, we have a fallback.

**Pull request merged**